### DISCUSSION TITLE
Extracting position-specific weight matrix from Bonito CRF model encoder

### DISCUSSION THREAD

**Discussion by AnnabelLarge** [nanoporetech/bonito#368] [Q&A]
**I have** Nanopore data that was collected with the older R10.3 pore. Thus, I have downloaded the Bonito weights corresponding with the Bonito CRF Model (the LSTM).

**I want** to extract something akin to a position-specific weight matrix, a (5xL) matrix that has the probabilities (or logits) over (A, T, G, C, [blank]) for each position in the output state path L. I believe this corresponds to the output of the encoder, before the beam decoding step.

**What I've tried so far:**
I suspect this is related to [line 34 of bonito/crf/basecall.py](https://github.com/nanoporetech/bonito/blob/f19c8c9a8fffec17822e9cd98cb7d7a93c4e2274/bonito/crf/basecall.py#L34) i.e. in the compute_scores.py function. For default parameters, `scores` is a (64, 800, 4096) matrix. From what I can tell, the dimensions are-

- (N) 64: default batch size is 32, multiply by 2 because bidirectional LSTM
- (T) 800: ???
- (C) 4096: default chunk size is 4096 (reads are divided up into overlapping chunks, processed, then stitched together at the end)

**My specific questions:** 
- Is this `scores` matrix somehow later interpreted as a PSWM? or is that coming from another part of bonito?
- If I'm looking at the correct matrix, then what is T?
- How does T (of size 800) get projected down to correspond to (A, T, G, C, [blank])? Does this happen somewhere in the ont-koi code?
